ote/src/cljs/taxiui/app/controller/front_page.cljs

```diff
@@ -0,0 +1,270 @@
+    (fn [app path]
+      (if (= menu-flag-path path)
+        (if (get-in app path) false true)
+        false))))
+
+(extend-protocol tuck/Event
+
+  ChangePage
+  (process-event [{given-page :given-page params :params :as e} app]
+    (navigate e app (fn [app]
+                      (do
+                        (routes/navigate! given-page params)
+                        app))))
+
+  GoToUrl
+  (process-event [{url :url :as e} app]
+    (navigate e app (fn [app]
+      (.setTimeout js/window #(set! (.-location js/window) url) 0)
+      app)))
+
+  OpenNewTab
+  (process-event [{url :url :as e} app]
+    (let [window-open (.open js/window)]
+         (set! (.-opener window-open) nil)
+         (set! (.-location window-open) url)
+    app))
+
+  StayOnPage
+  (process-event [_ app]
+    (dissoc app :navigation-prompt-open?))
+
+  ToggleFintrafficMenu
+  (process-event [_ app]
+    (toggle-menu app [:ote-service-flags :fintraffic-menu-open]))
+
+  ToggleMobileBottomMenu
+  (process-event [_ app]
+    ; mobile bottom menu is handled separately to support two-level nested menus
+    (assoc-in
+      app
+      [:ote-service-flags :mobile-bottom-menu-open]
+      (if (get-in app [:ote-service-flags :mobile-bottom-menu-open]) false true)))
+
+  ToggleUpdatesMenu
+  (process-event [_ app]
+    (toggle-menu app [:ote-service-flags :navigation-updates-menu]))
+
+  ToggleServiceInfoMenu
+  (process-event [_ app]
+    (toggle-menu app [:ote-service-flags :service-info-menu-open]))
+
+  ToggleMyServicesMenu
+  (process-event [_ app]
+    (toggle-menu app [:ote-service-flags :my-services-menu-open]))
+
+  ToggleSupportMenu
+  (process-event [_ app]
+    (toggle-menu app [:ote-service-flags :support-menu-open]))
+
+  ToggleUserMenu
+  (process-event [_ app]
+    (toggle-menu app [:ote-service-flags :user-menu-open]))
+
+  ToggleLangMenu
+  (process-event [_ app]
+    (toggle-menu app [:ote-service-flags :lang-menu-open]))
+
+  CloseHeaderMenus
+  (process-event [_ app]
+    (-> app
+        (close-all-menus)
+        ; mobile bottom menu is handled separately to support two-level nested menus
+        (assoc-in [:ote-service-flags :mobile-bottom-menu-open] false)))
+
+  Logout
+  (process-event [_ app]
+    (assoc-in app [:ote-service-flags :user-menu-open] true)
+    app)
+
+  EnsureTransportOperator
+  (process-event [_ app]
+     (if (:services-changed? app)
+      (get-transport-operator-data app)
+      app))
+
+  GetTransportOperator
+  (process-event [_ app]
+      (comm/post! "transport-operator/group" {} {:on-success (tuck/send-async! ->TransportOperatorResponse)
+                                                 :on-failure (tuck/send-async! ->TransportOperatorFailed)})
+      app)
+
+  TransportOperatorResponse
+  (process-event [{response :response} app]
+    (assoc app :transport-operator response))
+
+  TransportOperatorFailed
+  (process-event [{response :response} app]
+    ;; FIXME: figure out what the error is and add it to app state
```

Comment:
Just noticed this FIXME, it's also on `ote.app.controller.front-page`. I wonder if this is something that would need any action?

Reply:
Original is from November 2017 so maybe it's a non-issue...? :D

Reply:
Might be, I don't know what's the error here :D